exchange/exchange.go

```diff
@@ -587,6 +590,28 @@ func (e *exchange) makeBid(Bids []*pbsOrtbBid, adapter openrtb_ext.BidderName) (
 	return bids, errList
 }
 
+// If bid got cached inside `(a *auction) doCache(ctx context.Context, cache prebid_cache_client.Client, targData *targetData, bidRequest *openrtb.BidRequest, ttlBuffer int64, defaultTTLs *config.DefaultTTLs, bidCategory map[string]string)`,
+// a UUID should be found inside `a.cacheIds` or `a.vastCacheIds`. This function returns the UUID along with the internal cache URL
+func (e *exchange) getBidCacheInfo(bid *pbsOrtbBid, auc *auction) (openrtb_ext.ExtBidPrebidCacheBids, bool) {
+	var cacheInfo openrtb_ext.ExtBidPrebidCacheBids
+	var cacheUUID string
+	var found bool = false
+
+	if auc != nil {
+		if bid.bidType == openrtb_ext.BidTypeVideo {
```

Comment:
There may be use cases in the future for video bids to get a full bid cached, or non-video bids to get some sort of vast cache. Also we cannot 100% trust adapters to report the `bidType` correctly. We should just check for a bid cache first, and if that does not exist, check for a VAST cache.